contracts/bonding/BondingManager.sol

```diff
@@ -245,6 +249,11 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
             }
         }
 
+        if (del.delegateAddress == address(0)) {
+            // If no existing delegate, delegation amount = bonded stake + amount included in this function call
```

Comment:
If a delegator unbonds it will not have an existing delegate, but will still retain its bonded amount unless it waits through the unbonding period and calls `withdrawStake()`. If the delegator calls `bond()` with an amount of 0 it should expect to delegate its bonded amount to its target delegate. Previously, this wasn't the case and `bond()` would just delegate 0 to its target delegate because `bond()` did not handle the case where a delegator still has bonded stake but does not have a delegate. Now, this change will make sure to account for the delegator's existing bonded amount